From edbc944563484488068fc3d71add682bee7391c8 Mon Sep 17 00:00:00 2001
From: Henry Wang <Henry.Wang@arm.com>
Date: Sat, 8 Oct 2022 08:36:54 +0000
Subject: [PATCH] tools/xl-psr: Split display name for arch-specific
 terminologies

The xl-psr framework is currently supposed to be used for both
x86 "Intel Platform Shared Resource Monitoring/Control technologies"
and Arm MPAM extension, and some of the terminologies defined in
these two architectural extensions are different. For example,
the name of partition identifier in x86 is called COS (Class of
Service) while in Arm is called PARTID (Partition ID).

To display the correct terminologies on different architectures,
this commit splits the display name of these arch-specific
terminologies using per-arch marcos, and replace the original
x86-only wording in printf with the new per-arch name.

Signed-off-by: Henry Wang <Henry.Wang@arm.com>
Upstream-Status: Pending
Issue-Id: SCM-5244
---
 tools/xl/xl_psr.c | 26 +++++++++++++++++++-------
 1 file changed, 19 insertions(+), 7 deletions(-)

diff --git a/tools/xl/xl_psr.c b/tools/xl/xl_psr.c
index 7aff474d478f..afdfff2c3960 100644
--- a/tools/xl/xl_psr.c
+++ b/tools/xl/xl_psr.c
@@ -25,6 +25,14 @@
 #include "xl_utils.h"
 #include "xl_parse.h"
 
+#if defined(__i386__) || defined(__x86_64__)
+#define PARTITION_IDENTIFIER_NAME "COS"
+#define CAPACITY_BITMASK_NAME "CBM"
+#elif defined(__aarch64__)
+#define PARTITION_IDENTIFIER_NAME "PARTID"
+#define CAPACITY_BITMASK_NAME "CPBM"
+#endif
+
 static int psr_cmt_hwinfo(void)
 {
     int rc;
@@ -316,9 +324,10 @@ static int psr_l3_cat_hwinfo(void)
         printf("%-16s: %uKB\n", "L3 Cache", l3_cache_size);
         printf("%-16s: %s\n", "CDP Status",
                info[i].cdp_enabled ? "Enabled" : "Disabled");
-        printf("%-16s: %u\n", "Maximum COS", info[i].cos_max);
-        printf("%-16s: %u\n", "CBM length", info[i].cbm_len);
-        printf("%-16s: %#llx\n", "Default CBM",
+        printf("%-16s: %u\n", "Maximum "PARTITION_IDENTIFIER_NAME,
+               info[i].cos_max);
+        printf("%-16s: %u\n", CAPACITY_BITMASK_NAME" length", info[i].cbm_len);
+        printf("%-16s: %#llx\n", "Default "CAPACITY_BITMASK_NAME,
                (1ull << info[i].cbm_len) - 1);
     }
 
@@ -437,12 +446,14 @@ static int psr_print_socket(uint32_t domid,
             printf("%-16s: %uKB\n", "L3 Cache", l3_cache_size);
         }
 
-        printf("%-16s: %#llx\n", "Default CBM",
+        printf("%-16s: %#llx\n", "Default "CAPACITY_BITMASK_NAME,
                (1ull << info->u.cat.cbm_len) - 1);
         if (info->u.cat.cdp_enabled)
-            printf("%5s%25s%16s%16s\n", "ID", "NAME", "CBM (code)", "CBM (data)");
+            printf("%5s%25s%16s%16s\n", "ID", "NAME",
+                   CAPACITY_BITMASK_NAME" (code)",
+                   CAPACITY_BITMASK_NAME" (data)");
         else
-            printf("%5s%25s%16s\n", "ID", "NAME", "CBM");
+            printf("%5s%25s%16s\n", "ID", "NAME", CAPACITY_BITMASK_NAME);
 
         break;
     }
@@ -623,7 +634,8 @@ static int psr_mba_hwinfo(void)
         printf("Socket ID               : %u\n", info[i].id);
         printf("Linear Mode             : %s\n",
                info[i].u.mba.linear ? "Enabled" : "Disabled");
-        printf("Maximum COS             : %u\n", info[i].u.mba.cos_max);
+        printf("%-24s: %u\n", "Maximum "PARTITION_IDENTIFIER_NAME,
+               info[i].u.mba.cos_max);
         printf("Maximum Throttling Value: %u\n", info[i].u.mba.thrtl_max);
         printf("Default Throttling Value: %u\n", 0);
     }
-- 
2.34.1

